src/main/java/de/appelgriepsch/logback/GelfAppender.java

```diff
@@ -81,12 +81,10 @@ protected void append(ILoggingEvent event) {
         }
 
         if (includeMDC) {
-            for (Map.Entry<String, String> entry : event.getMDCPropertyMap().entrySet()) {
-                builder.additionalField(entry.getKey(), entry.getValue());
-            }
+            this.additionalFields.putAll(event.getMDCPropertyMap());
```

Comment:
I think this change introduces a bug. Previously the dynamic additional fields where only added to the builder, not to the appenders `this.additionalFields` (these are only the additional fields from the configuration file).

Say a log event comes with property `foo` included in the MDC, than this property is added to `this.additionalFields` and all those additional fields are included in the builder. That's fine so far. But when there's a second log event without the property `foo`, then this event's gelf message should not include `foo` (but it will with this change).

Reply:
Oh, good catch. Working on fixing this now.